testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/PasswordPage.java

```diff
@@ -46,8 +49,20 @@ public String getPassword() {
         return passwordInput.getAttribute("value");
     }
 
+    public String getPasswordError() {
+        try {
+            return passwordError != null ? passwordError.getText() : null;
+        } catch (NoSuchElementException e) {
+            return null;
+        }
+    }
+
     public String getError() {
-        return loginErrorMessage != null ? loginErrorMessage.getText() : null;
+        try {
+            return loginErrorMessage != null ? loginErrorMessage.getText() : getPasswordError();
```

Comment:
The test should be deterministic and should know whether it's fetching the error message from the notification or from the field.